source/DD4T.Templates.Base/Utils/EclProcessor.cs

```diff
@@ -231,6 +264,27 @@ private static FieldSet CreateExternalMetadataFieldSet(IEnumerable<IFieldDefinit
             return fieldSet;
         }
 
+        private static void CreateExternalMetadataXmlAttribute(IEnumerable<IFieldDefinition> eclFieldDefinitions, XmlElement parentElement, XmlElement xlinkElement)
+        {
+            foreach (IFieldDefinition eclFieldDefinition in eclFieldDefinitions)
+            {
+                XmlNodeList fieldElements = parentElement.SelectNodes(string.Format("*[local-name()='{0}']", eclFieldDefinition.XmlElementName));
+                if (fieldElements.Count == 0)
+                {
+                    // Don't generate a DD4T Field for ECL field without values.
+                    continue;
+                }
+
+                foreach (XmlElement fieldElement in fieldElements)
+                {
+                    if (fieldElement.HasChildNodes)
+                    {
+                        xlinkElement.SetAttribute(string.Format("data-{0}", eclFieldDefinition.XmlElementName), fieldElement.InnerText);
```

Comment:
Wouldn't it be better to always identify these data attributes as 'data-ecl-...'? Otherwise they could conflict with other data attributes.